### DISCUSSION TITLE
Regular expression pattern matching is possible in github actions?

### DISCUSSION THREAD

**Discussion by yogesh-kumitisomu-by** [community/community#26771] [Actions]
<p>Is regular expression pattern matching is possible in Github actions?</p>
<p>I have few release branches with name release/v1.1, release/v1.2 etc…</p>
<p>Let’s say my current branch is release/v1.1. How do i pass it as a regular expression with the pattern that matches my branch ?</p>
<p>if: contains(github.ref, ‘release/*’)</p>
<p>or</p>
<p>if [[  ${{ env.BRANCH }} =~ release/.* ]]</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="yogesh-kumitisomu-by" data-post="1" data-topic="209756">
<div class="title">
<div class="quote-controls"></div>
 yogesh-kumitisomu-by:</div>
<blockquote>
<p>if: contains(github.ref, ‘release/*’)</p>
</blockquote>
</aside>
<p><code>contains()</code> is an Actions function, but it doesn’t support regular expressions. If you just want to know if the branch name starts with <code>release/</code>, you could do something like this:</p>
<pre><code class="lang-yaml">if: startsWith(github.ref, 'refs/heads/release/')
</code></pre>
<aside class="quote no-group" data-username="yogesh-kumitisomu-by" data-post="1" data-topic="209756">
<div class="title">
<div class="quote-controls"></div>
 yogesh-kumitisomu-by:</div>
<blockquote>
<p>if [[ ${{ env.BRANCH }} =~ release/.* ]]</p>
</blockquote>
</aside>
<p>That is Bash syntax, you can use it within a <code>run</code> step if using Bash as the shell (default on Linux).</p>

**Comment by yogesh-kumitisomu-by** [comment]
<p>Somehow the below condition doesn’t work for me if i use it in run step.</p>
<p>if [[ ${{ env.BRANCH }} =~ release/.* ]]</p>
<p>Is it the right way to declare ?</p>

**Comment by airtower-luna** [comment]
<p>You’ll have to provide more context to answer that, ideally the full workflow and a description of <em>how</em> it doesn’t work, like in a bug report: What do you expect, what actually happens?</p>
<p>And of course that only works if you actually <em>have</em> <code>BRANCH</code> environment variable. :slightly_smiling_face:</p>

**Comment by yogesh-kumitisomu-by** [comment]
<p>Thanks. I tried with the below one and it worked.</p>
<p>if: startsWith(github.ref, ‘refs/heads/release/’)</p>
<p>Considering base branch as release, how can i check for the pull request level ?</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="yogesh-kumitisomu-by" data-post="5" data-topic="209756">
<div class="title">
<div class="quote-controls"></div>
 yogesh-kumitisomu-by:</div>
<blockquote>
<p>Considering base branch as release, how can i check for the pull request level ?</p>
</blockquote>
</aside>
<p>I’m not sure what you mean, could you explain? Something related to merging into a release branch?</p>

**Comment by yogesh-kumitisomu-by** [comment]
<p>Ok. Let’s say i have multiple release branches with the name release/v1.1, release/v1.2 (This is at the branch level the below condition works</p>
<p>if: startsWith(github.ref, ‘refs/heads/release/’)</p>
<p>Similarly if developer raises a Pull request for any of these release branches release/v1.1 or release/v1.2. I know something like below will work for exact matching branch…How can i have the if condition to check for multiple pull request release branches ?</p>
<p>if: github.event.pull_request.base.ref == ‘release/v1.1’<br>
env:<br>
LABEL: test</p>
<p>Because based on the above PR for the respective release branches, i need to set an environment variable. But for all the release branches i am going to use the same environment variable as test.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>One option would be to use <code>startWith()</code> on <code>github.event.pull_request.base.ref</code>.</p>
<p>If you need more complex logic the best thing might be to feed the base ref to a custom command/script, and decide what to write to <code>GITHUB_ENV</code> in there. That way you have a lot more options to write the logic you need.</p>

**Comment by yogesh-kumitisomu-by** [comment]
<p>Thanks.  This helps!!!</p>
